### DISCUSSION TITLE
Detect caps lock when typing master password

### DISCUSSION THREAD

**Discussion by geofferb** [bitwarden/clients#8147] [Password Manager]
### Code Contribution Proposal

I would like to take a crack at this [long requested feature](https://community.bitwarden.com/t/detect-caps-lock-when-typing-master-password-and-notify-user-in-the-ui/3944) for my first contribution here. 

So far, I've found that the master password login page is handled by [lock.component.ts](https://github.com/bitwarden/clients/blob/main/apps/browser/src/auth/popup/lock.component.ts) and [lock.component.html](https://github.com/bitwarden/clients/blob/main/apps/browser/src/auth/popup/llock.component.html) in all three clients. I plan to implement something along the lines of [this solution](https://stackoverflow.com/a/45035806/14659292), but I'm not sure what the notification would look like; perhaps something like this private mode warning?
![image](https://github.com/bitwarden/clients/assets/38046772/9da030a6-d01e-4ab9-8358-2dce565f4bb0)

I would appreciate any guidance on how to proceed with this feature, thanks in advance.

**Comment by djsmith85** [comment]
@geofferb Thank you for your interest in contributing to Bitwarden. I have passed this onto the Product and Design teams and they will be able to offer guidance on how to inform a user about caps lock being active.

Kind regards,
Daniel

**Comment by BetoMees** [comment]
This is the simple way I could think
![image](https://github.com/bitwarden/clients/assets/15331041/435455f0-a80e-4b63-9dc5-d6bedb6d1a96)

But you could just use an icon like this example:
![image](https://github.com/bitwarden/clients/assets/15331041/ba135670-2763-495d-96b1-2f3fdf12f731)
With the icon you will have to include a new glyph into `libs\angular\src\scss\bwicons\fonts\bwi-font.svg` and `libs\angular\src\scss\bwicons\styles\style.scss`

**Comment by kenny9p** [comment]
Just realized that this isn't a thing yet

**Comment by Speculate7348** [comment]
This is needed.
